src/vm/database/clarity_db.rs

```diff
@@ -680,6 +716,41 @@ impl<'a> ClarityDatabase<'a> {
             .expect("Failed to get block data.")
     }
 
+    /// Gets the `StacksBlockId` for the *current* chain tip Stacks block height, converts this
+    /// `StacksBlockId` to a `SortitionId` and then gets the `BurnchainHeaderHash` from the
+    /// `BurnStateDB` using this `SortitionId`.
+    ///
+    /// Returns Some() if `0 <= height < current_burnblock_height`, and None otherwise.
+    pub fn get_burnchain_block_header_hash_for_burnchain_height(
+        &mut self,
+        burnchain_block_height: u32,
+    ) -> Option<BurnchainHeaderHash> {
+        let current_stacks_height = self.get_current_block_height();
+        let id_bhh = self.get_index_block_header_hash(current_stacks_height);
+
+        // Is this block in the database?
+        // Should this return an error/panic if it fails?
+        let consensus = match self.headers_db.get_consensus_hash_for_block(&id_bhh) {
+            None => {
+                return None;
```

Comment:
Should these None cases 1) propagate None, 2) return Error, 3) panic?

Reply:
It should panic. _Except_ in the case that the current block height is 0. That is, a parent block should _always_ have a consensus hash (unless it's at height 0 and there is no parent).